osquery/utils/conversions/windows/strings.cpp

```diff
@@ -88,13 +88,15 @@ std::string bstrToString(const BSTR src) {
 LONGLONG cimDatetimeToUnixtime(const std::string& src) {
   // First init the SWbemDateTime class
   ISWbemDateTime* pCimDateTime = nullptr;
+  auto pCimDateTimeManager =
+      scope_guard::create([&pCimDateTime]() { pCimDateTime->Release(); });
```

Comment:
To be extra safe I would write 

```cpp
if (pCimDateTime != nullptr) {
  pCimDateTime->Release();
}
```

as the lambda expression.

I recommend this because I worry about the situation where line 94 errors early and does not initialize `pCimDateTime`.

Reply:
Reasonable looking safety, I updated it. Though with the reordering Zach suggested I don't know if it's needed